src/Microsoft.Framework.PackageManager/Utils/LockFileUtils.cs

```diff
@@ -173,6 +175,17 @@ public static LockFileTargetLibrary CreateLockFileTargetLibrary(LockFileLibrary
                 }
             }
 
+            if (!hasCompatibleDependencySet &&
```

Comment:
To be clear, @davidfowl means `_._` (markdown formatting got in the way), but yes, I would actually just detect  the case where `_._` is the sole entry in any of the lists and remove it, resulting in a zero-length array.